line_detection_updated.py

```diff
@@ -0,0 +1,302 @@
+        self.image = None
+
+        # Image processing parameters
+        self.mask = False
+        self.mask_percent = 0.2
+        self.nstd1_cut = 2
+        self.nstd2_binary_cut = 0.5
+
+        # Line detection parameters
+        self.threshold = 0.075
+
+        # Blurring parameters
+        self.flux_prop_thresholds = [0.1, 0.2, 0.3, 1]
+        self.blur_kernel_sizes = [3, 5, 9, 11]
+
+    def process_image(self):
+        '''
+        Processes the raw image for hough_lines
+
+        Parameters
+        ----------
+        self : 'class'
+            The LineDetection class to use its __init__ parameters
+
+        Returns
+        -------
+        thresholded_image: 'numpy.ndarray'
+            The image after applying brightness cuts and thresholding on
+            the raw image
+        blurred_image: 'numpy.ndarray'
+            The image after applying blurring to the thresholded image
+        edges: 'numpy.ndarray'
+            The Canny edge detected image on which hough_lines would
+            be applied
+        '''
+        trimmed_image = self.image
+
+        # Making first cuts in the image
+        # Outliers on the positive side take the value of the cut.
+        # Outliers on the negative side take the value 0
+        up_limit = trimmed_image.mean() + self.nstd1_cut*trimmed_image.std()
+        low_limit = trimmed_image.mean() - self.nstd1_cut*trimmed_image.std()
+        trimmed_image[trimmed_image > up_limit] = up_limit
+        trimmed_image[trimmed_image <= low_limit] = 0
+
+        # Standardizing the image and moving it away from 0
+        processed_image = (trimmed_image-trimmed_image.mean())/trimmed_image.std()
+        processed_image -= processed_image.min()
+
+        # Fitting the image to a 0-255 range
+        processed_image = (255*(processed_image - processed_image.min())
+                           / (processed_image.max() - processed_image.min()))
+
+        # Thresholding the processed image
+        limit = processed_image.mean() + self.nstd2_binary_cut*processed_image.std()
+        threshold, thresholded_image = cv2.threshold(processed_image, limit, 255, cv2.THRESH_BINARY)
+        thresholded_image = cv2.convertScaleAbs(thresholded_image)
+
+        # Masking image to remove any borders
+        if self.mask is True:
+            x_top, x_bottom, y_left, y_right = image_mask(thresholded_image, self.mask_percent)
+
+            thresholded_image = thresholded_image[x_top:x_bottom, y_left:y_right]
+
+        # Deciding Kernel size of blur based on amount of noise
+        num_bright_pixels = np.sum(thresholded_image > np.mean(thresholded_image))
+        x = thresholded_image.shape[0]
+        y = thresholded_image.shape[1]
+        prop_bright_pixels = num_bright_pixels/(x*y)
+
+        for i in range(len(self.flux_prop_thresholds)):
+            if prop_bright_pixels < self.flux_prop_thresholds[i]:
+                kernel_blur = self.blur_kernel_sizes[i]
+                break
```

Comment:
This is a nice solution to having a dynamic range of blur kernel parameters and thresholds.